src/users/users.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable,HttpException, HttpStatus, ConflictException, UnauthorizedException, NotFoundException } from '@nestjs/common';
```

Comment:
Overall, the code seems to be well-written and follows the Nest.js best practices. However, it's always a good idea to perform thorough testing to ensure the code behaves as expected in different scenarios.